warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlStringBuildingVisitor.java

```diff
@@ -154,14 +196,14 @@ public Object visit(ASTAndNode node, Object data) {
             sb.append("(");
         }
         
-        int lastsize = sb.length();
+        Collection<String> childStrings = (sortDedupeChildren) ? new TreeSet<>() : new ArrayList<>();
```

Comment:
The ArrayList could take the number of child nodes as an argument.

Reply:
Sure.  See latest commit.